lib/puppet/application/filebucket.rb

```diff
@@ -148,6 +158,29 @@ def restore
     @client.restore(file, md5)
   end
 
+  def diff
+    checksum_a = args.shift
+    checksum_b = args.shift
+    file_a = nil
+    file_b = nil
+    if checksum_a && Puppet::FileSystem.exist?(checksum_a)
+      # It's a file
+      file_a = checksum_a
+      checksum_a = nil
+    end
+    if checksum_b && Puppet::FileSystem.exist?(checksum_b)
+      # It's a file
+      file_b = checksum_b
+      checksum_b = nil
+    end
+    if (checksum_a || file_a) && (checksum_b || file_b)
+      Puppet.info("Comparing #{checksum_a} #{checksum_b} #{file_a} #{file_b}")
+      print @client.diff(checksum_a, checksum_b, file_a, file_b)
+    else
+      raise Puppet::Error, "Need at least two arguments: filebucket diff #{ARGV} <file_a> <file_b>"
```

Comment:
The `ARGV` here prints out a weird error message, in my opinion:

`Error: Could not run: Need at least two arguments: filebucket diff ["filebucket", "diff"] <file_a> <file_b>`

I think we could remove `#{ARGV}` here as it's confusing.